### ISSUE TITLE
Show Image in the wodget

### ISSUE THREAD

**Issue by atif2148** [ABausG/home_widget#91]
Is it possible to show a network image from rest api? If yes can you please specify a code for it?
If no, till when can we expect this functionality to be added?
Regards

**Comment by Viktohblake** [comment]
> Is it possible to show a network image from rest api? If yes can you please specify a code for it? If no, till when can we expect this functionality to be added? Regards

Have you been able to do this? i'm also having issues showing image in the widget

**Comment by milindgoel15** [comment]
So have you find any way to do this?

I followed up the same steps in [google codelab](https://codelabs.developers.google.com/flutter-home-screen-widgets#5)s to render it but came across an error. We can render full flutter widget saved as a screenshot in the imageview. 
```dart
    var path = await HomeWidget.renderFlutterWidget(
      Image.network(
        "https://openweathermap.org/img/wn/${data.currentData.icon}@4x.png",
        height: 64,
        width: 64,
      ),
      logicalSize: const Size(64, 64),
      key: "weather_image",
    );
    print(path); 
```
But gives

```logs
Exception has occurred.
_Exception (Exception: Failed to render the widget: Exception: Failed to save screenshot to app group container: Null check operator used on a null value)
```

**Comment by ABausG** [comment]
Yes with Network Images inside the `renderFlutterWidget` it would need to wait for the Network Image to be loaded. My plan is to provide a similar method of saving an Image to a path in the future.
For now you could try to load the image as byte data. Convert it to base64 and then save that as a string

**Comment by imgarvish** [comment]
Why to convert the byte data to base64 string?

**Comment by milindgoel15** [comment]
Can we also save svg images through flutter_svg package?

i am trying this way:
```dart
    var path = await HomeWidget.renderFlutterWidget(
      SvgPicture.asset(
        "assets/icons/$weatherSource/${data.currentData.icon}.svg",
        height: 64,
        width: 64,
        alignment: Alignment.center,
      ),
      logicalSize: const Size(64, 64),
      key: "weather_image",
    );
```
But i get same error as it did with network image.

**Comment by milindgoel15** [comment]
Just an update, I have created a pull request to fix the null pointer exception. 

I believe all types of images will now work. I have already tested SVG, network image, and png file asset and it works fine on my end.

**Comment by esDotDev** [comment]
> Yes with Network Images inside the `renderFlutterWidget` it would need to wait for the Network Image to be loaded. My plan is to provide a similar method of saving an Image to a path in the future. For now you could try to load the image as byte data. Convert it to base64 and then save that as a string

Any idea on when you might add this feature? We have a similar need to somehow show a network image in the widget. 

Are there any code examples for the Base64 approach?

**Comment by esDotDev** [comment]
Interesting, do you need to do anything to deal with the asynchronous download time? Or can you just render NetworkImage directly?

**Comment by milindgoel15** [comment]
It doesn't require any extra configurations, just declare the image.network as a widget and pass it to renderFlutterWidget. 

I didn't need to add any async code to handle downloading time as flutter handled it automatically.

**Comment by esDotDev** [comment]
I ended up trying the base64 approach and it seems to be working well.

In flutter:
```dart
final bytes = await http.readBytes(Uri.parse(imageUrl));
final imageBase64 = base64Encode(bytes);
HomeWidget.saveWidgetData<String>('lastDiscoveredImageData', imageBase64);
```

In swiftUi:
```swift
let userDefaults = UserDefaults(suiteName: "...")
let imageData = userDefaults?.string(forKey: "lastDiscoveredImageData") ?? ""
if(imageData.isEmpty == false) {
    let uiImage = UIImage(data: Data(base64Encoded: entry.imageData)!)
    let image = Image(uiImage: uiImage!);
    ....
}
```

**Comment by imgarvish** [comment]
for anyone who is still wondering how to load the image using `Image.network()` widget.

here's an example code i used:
```
    Image downloadImage = Image.network(
      "https://storage.googleapis.com/cms-storage-bucket/c823e53b3a1a7b0d36a9.png",
      fit: BoxFit.contain,
      height: 1080,
      width: 1080,
    );

    final ImageStream stream =
        downloadImage.image.resolve(ImageConfiguration.empty);

    final Completer<void> completer = Completer<void>();
    stream.addListener(ImageStreamListener((image, synchronousCall) {
      completer.complete();
    }));

    await completer.future;

    final result = await HomeWidget.renderFlutterWidget(
        SizedBox(height: 1080, width: 1080, child: downloadImage),
        key: "custom-flutter-widget",
        logicalSize: const Size(1080, 1080),
        pixelRatio: WidgetsBinding
            .instance.platformDispatcher.views.first.devicePixelRatio);

    debugPrint(result);
```

also make sure to use @milindgoel15's PR. see that here: https://github.com/ABausG/home_widget/pull/155.

**Comment by Rart3001** [comment]
Hello, do you have a solution for this case?

**Comment by tiyberius** [comment]
Oh my GOSH thank you @Rart3001 for your solution. The LLM's came up with a nasty solution, but yours is way simpler and actually works. Lifesaver. Thank you!! 🙏🏻

**Comment by ABausG** [comment]
Once #409 is merged you can do: `HomeWidget.saveImage(NetworkImage(url))` to store the image :)
